### ISSUE TITLE
`RUF012` shouldn't complain when type hint is not present

### ISSUE THREAD

**Issue by harshil21** [astral-sh/ruff#5244]
The following snippet produces an error which I think it shouldn't since there is no type annotation for that variable:

```python
# test.py
class A:
    var = ["hey", "there"]
```

This gets particularly annoying in test suites, since adding type hints there (specially in this case) is not required imo.

### Command

`ruff test.py --select RUF --isolated`

```
tests/ruff.py:2:11: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
Found 1 error.
```

### Version

`ruff 0.0.274`

**Comment by scastlara** [comment]
related https://github.com/astral-sh/ruff/issues/5243

**Comment by harshil21** [comment]
yeah just opened within a minute of each other :D

**Comment by charliermarsh** [comment]
(Closing as a duplicate of #5244.)

**Comment by Thomasdezeeuw** [comment]
> (Closing as a duplicate of #5244.)

I think you meant https://github.com/astral-sh/ruff/issues/5243, since this is #5244.
